### PR TITLE
add storybook

### PR DESCRIPTION
**Pull request by mattapperson**
added storybook and the main components to a storybook

### CODE REVIEW

**Comment by sorenlouv** [comment]
@mattapperson The tests are failing on my end `npm test`. Same for you?

File: src/Histogram/customplot.stories.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import { storiesOf, linkTo } from '@storybook/react';
 import d3 from 'd3';
```
Comment (sorenlouv) [inline src/Histogram/customplot.stories.js]: This file should still be called histogram.stories.js, right?

File: src/Legend/legend.stories.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { storiesOf, linkTo } from '@storybook/react';
```
Comment (sorenlouv) [inline src/Legend/legend.stories.js]: `linkTo` is unused

File: src/Timeline/timeline.stories.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { storiesOf, linkTo } from '@storybook/react';
```
Comment (sorenlouv) [inline src/Timeline/timeline.stories.js]: `linkTo` is unused

File: src/CustomPlot/customplot.stories.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import { storiesOf, linkTo } from '@storybook/react';
```
Comment (sorenlouv) [inline src/CustomPlot/customplot.stories.js]: `linkTo` is unused

File: src/Histogram/customplot.stories.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import { storiesOf, linkTo } from '@storybook/react';
```
Comment (sorenlouv) [inline src/Histogram/customplot.stories.js]: `linkTo` is unused

File: src/Timeline/timeline.stories.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
```
Comment (sorenlouv) [inline src/Timeline/timeline.stories.js:1]: Should Timeline/index.js perhaps be deleted, now we have this file?

File: src/Legend/legend.stories.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (sorenlouv) [inline src/Legend/legend.stories.js:1]: Same as with Timeline: should Legend/index.js be deleted?

**Comment by mattapperson** [comment]
@sqren they were failing for me before I added storybook so I paid little attention to that honestly...

**Comment by sorenlouv** [comment]
Ah, okay. I must have broken them recently then (reminds me that I should setup travis...)

**Comment by sorenlouv** [comment]
@mattapperson If I checkout master they fail - but if I delete node_modules and run them they work. So it seems to be something in the storybook changes. Can you reproduce this, or is it just my setup?

**Comment by sorenlouv** [comment]
(I just updated a snapshot on master - the tests should be passing again)

File: .storybook/config.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+/* eslint-env node */
```
Comment (sorenlouv) [inline .storybook/config.js]: This shouldn't be necessary: https://github.com/sqren/apm-components/blob/master/.eslintrc.json#L12

**Pull request merged**